tt_metal/impl/device/device.cpp

```diff
@@ -1458,19 +1458,6 @@ void Device::compile_command_queue_programs() {
     std::string prefetch_kernel_path = "tt_metal/impl/dispatch/kernels/cq_prefetch.cpp";
     std::string dispatch_kernel_path = "tt_metal/impl/dispatch/kernels/cq_dispatch.cpp";
 
-    // TODO: These are semaphore IDs, remove these when CreateSemaphore returns ID rather than address
```

Comment:
Nice, finally getting this cleaned up. Make sure you run T3k/TG tests, I think there is some risk of regression here, since the semaphore ids are returned based on first available.